internal/grpc/constants.go

```diff
@@ -0,0 +1,7 @@
+package grpc
+
+type contextKey string
+
+const (
+	ClientCNKey = contextKey("clientCN")
```

Comment:
Consider naming this `ClientIDKey` or similar.

"CN" is a concept of the certificate, but now that we've taken the identity out of the certificate and stuffed it in the context the rest of the code doesn't really care that it originally came from a cert's common name.